### PR TITLE
Implementing new components.

### LINKED ISSUE #7: Implement new circuit elements
We only have 4 circuit elements. We need to implement more elements.

### PR DESCRIPTION
**Pull request by JeanLColombo**
Implementation of noveau components to Pamola's library.

### CODE REVIEW

**Comment by JohniMichels** [comment]
Can we add a test to see if it's behaving correctly?

**Comment by JeanLColombo** [comment]
> Can we add a test to see if it's behaving correctly?

Yes, already done.

**Comment by JeanLColombo** [comment]
If `G.Terminal` is already connected to a node, the following code doesn't work.

` G.Terminal.ConnectTo(J.Positive);`

But this one works:

` G.Terminal.Node.ConnectTo(J.Positive);`

Was this an implementational feature, or a bug? Because Node will only exist after being created, so the second line may end in runtime errors.

**Comment by JohniMichels** [comment]
> If `G.Terminal` is already connected to a node, the following code doesn't work.
> 
> ` G.Terminal.ConnectTo(J.Positive);`
> 
> But this one works:
> 
> ` G.Terminal.Node.ConnectTo(J.Positive);`
> 
> Was this an implementational feature, or a bug? Because Node will only exist after being created, so the second line may end in runtime errors.

It's a feature, to ensure explicit triple connection. You can use the ConnectAll method to ease the verbosity.

File: Source/Core/Components/Jumper.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    public class Jumper : Dipole
+    {
+        /// <summary>
+        /// Returns the electric behavior of a jumper.
+        /// </summary>
+        /// <returns>When both terminals are connected, and the nodes created, the voltage
+        /// drop in a jumper is zero.</returns>
+        private Complex JumperBehavior()
+        {
+            if (!Positive.IsConnected() || !Negative.IsConnected()) return new Complex();
```
Comment (JohniMichels) [inline Source/Core/Components/Jumper.cs]: Can we use a ternary operation here?

```
return (Positive.IsConnected() && Negative.IsConnected()) ?
Positive.Node.Voltage-Negative.Node.Voltage :
new Complex()
```

**Comment by JeanLColombo** [comment]
> > If `G.Terminal` is already connected to a node, the following code doesn't work.
> > ` G.Terminal.ConnectTo(J.Positive);`
> > But this one works:
> > ` G.Terminal.Node.ConnectTo(J.Positive);`
> > Was this an implementational feature, or a bug? Because Node will only exist after being created, so the second line may end in runtime errors.
> 
> It's a feature, to ensure explicit triple connection. You can use the ConnectAll method to ease the verbosity.

image

This works.

**Review by JohniMichels — DISMISSED** [review]

File: Source/Core/Components/Jumper.cs (hunk 1 above)
Comment (JeanLColombo) [reply Source/Core/Components/Jumper.cs]: Yep. I think it's doable. 

Will it enhance performance, perhaps?

**Review by JeanLColombo — COMMENTED** [review]
Were we able to acquire the data properly?

How long did it take to get the simulation results?

**Review by JohniMichels — APPROVED** [review]
LGTM

**Pull request merged**